overlord/fdestate/fdemgr.go

```diff
@@ -472,6 +472,104 @@ func GetKeyslots(st *state.State, keyslotRefs []KeyslotRef) (keyslots []Keyslot,
 	return mgr.GetKeyslots(keyslotRefs)
 }
 
+func keyslotsAffectedByTask(t *state.Task) ([]KeyslotRef, error) {
```

Comment:
I think this is a bit magic as it expects all the tasks do use "keyslots" properly. If we add more tasks modifying keyslots, it is not always obvious that this is a pattern that is required. Not sure how to improve that though. But maybe something to keep in mind and figure out if we can make it more obvious.